### ISSUE TITLE
[no-unsafe-call] False positive when using dynamic import

### ISSUE THREAD

**Issue by jraoult** [typescript-eslint/typescript-eslint#1717]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-call": "error"
  }
}
```

```TS
const promise = import("typedash");
```

**Expected Result**
Pass the test since import is not an `any` (cf. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-4.html#dynamic-import-expressions)

**Actual Result**
Reports  _Unsafe call of an any typed value_

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.23.0` |
| `@typescript-eslint/parser`        | `2.23.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.1` |

**Comment by bradzacher** [comment]
hey! someone that's using the new no-unsafe rules!
Off topic, what do you think of them?

---

Interesting.
This is just due to typescript's weird handling of its keyword functions.

It treats the `import` keyword as type `any`, but then it has special typing for the output of the `import`
The same thing happens with the `const` keyword (`const x = 1 as const`).

I knew about the latter, but not the former.

You can see the behaviour by clicking on either keywords in this ts-ast-viewer repro:
https://ts-ast-viewer.com/#code/MYewdgzgLgBADgJxAWwJYQKYwLw1cuEBKACgCIoBPODAEwEMIALMgSgG4AoUSWADxwwAjDEYwe0LkA

I'll look to adding some special handling for this.

**Comment by jraoult** [comment]
> hey! someone that's using the new no-unsafe rules!
Off topic, what do you think of them?

@bradzacher they are brilliant. This is the only way to make sure there are no hidden `any`s since it is impossible to see otherwise. It should be a recommended set of rules if you ask me. Thx a lot for them.

**Comment by scvnathan** [comment]
I just started introducing these rules to our codebase and ran into the same issue. These are great rules for us because we have a huge mixed js/ts codebase and have `noImplicitAny` turned off while we slowly migrate. We also make heavy use of [loadable-components](http://loadable-components.com) which is where I ran into this issue.
